ModelMetadataExtensions/ConventionalModelMetadataProvider.cs

```diff
@@ -92,10 +92,11 @@ public ConventionalModelMetadataProvider(bool requireConventionAttribute, Type d
         private static void ApplyConventionsToValidationAttributes(IEnumerable<Attribute> attributes, Type containerType,
             string propertyName, Type defaultResourceType)
         {
-            foreach (
-                ValidationAttribute validationAttribute in attributes.Where(a => (a as ValidationAttribute != null)))
+            foreach (var validationAttribute in attributes.OfType<ValidationAttribute>())
             {
-                if (string.IsNullOrEmpty(validationAttribute.ErrorMessage))
+                if (string.IsNullOrEmpty(validationAttribute.ErrorMessage)
+                    || validationAttribute is DataTypeAttribute
+                    && validationAttribute.ErrorMessage == ((ValidationAttribute)Activator.CreateInstance(validationAttribute.GetType())).ErrorMessage)
```

Comment:
It's conceivable that `CreateInstance` could throw an exception. To be safe, we should wrap that in a try/catch.

Reply:
Yeap, I agree, moved extraction of the defaultErrorMessage to a separated method wrapped in try/catch.